python/mmSolver/utils/raytrace.py

```diff
@@ -92,8 +93,20 @@ def closest_intersect(source,
     min_dist = max_dist
     for point in hit_points:
         dist = source_pt.distanceTo(point)
+        dot = (point.x * source_pt.y) - (point.y * source_pt.x)
```

Comment:
This is not how a dot product is computed.
This pseudo code shows the calculation (in 3D space).
```
dot_ab = (a.x × b.x) + (a.y × b.y) + (a.z x b.z)
```

However, a dot product can be computed using Maya's MVector (or MFloatVector) classes (search on the page for "dot product"):
http://help.autodesk.com/view/MAYAUL/2018/ENU/?guid=__cpp_ref_class_m_vector_html

Therefore you'll need to get directions (vectors) to compare. `direction_vec` already exists, that can be one of the vectors for the dot product. Next you'll need to get a direction from source_pt to `hit_pt`, you can do that with something similar to `mmSolver.tools.raycastmarker.tool.__get_camera_direction_to_point`.